File: app/models/teacher.rb
class Teacher < ActiveRecord::Base
  has_many :students
  has_many :worksheets
  has_many :responses, through: :students

  validates :first_name, :last_name, :class_code, :password, :password_confirmation, presence: true
  validates :email, presence: true, uniqueness: true
  validates_format_of :email, :with => /\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\z/i

  has_secure_password

  before_validation :downcase_fields
  def downcase_fields
    self.email = self.email.downcase
  end
end

File: app/assets/javascripts/models/teacher.js
App.Teacher = DS.Model.extend({
  firstName: DS.attr('string'),
  lastName: DS.attr('string'),
  email: DS.attr('string'),
  classCode: DS.attr('string'),
  worksheets: DS.hasMany('worksheet', {
    async: true
  })
})

App.Teacher.FIXTURES = [{
  id: 1,
  firstName: 'Bill',
  lastName: 'Bill',
  email: 'bill@bill.com',
  school: 1
}]
